### PR TITLE
chore: parse & map repository errors

### PR DESCRIPTION
**Pull request by vindard**
## Description

This is to map known connection issues to take the place of this class of `UnknownRepositoryError` errors.

### CODE REVIEW

**Review by bodymindarts — DISMISSED** [review]

File: src/services/shared/repository.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { CantConnectToMongoDbError, UnknownRepositoryError } from "@domain/errors"
```
Comment (bodymindarts) [inline src/services/shared/repository.ts:1]: Shouldn't this be scoped under `mongoose`? its not a service in itself.

Comment (vindard) [reply src/services/shared/repository.ts]: Oh yea, the vast majority of our repository implementations are already in there anyway.

There's a few places outside of `mongoose` where we call `import { parseRepositoryError } from "@services/mongoose/utils"`. Should we expose those `utils` at the package level, or is it ok to reach in like this?

File: src/domain/errors.ts [hunk 2]
```diff
@@ -13,6 +13,9 @@ export class DuplicateError extends RepositoryError {}
 
 export class BadInputsForFindError extends RepositoryError {}
 export class CouldNotFindError extends RepositoryError {}
+export class CantConnectToMongoDbError extends RepositoryError {
```
Comment (bodymindarts) [inline src/domain/errors.ts:16]: shouldn't couple domain to `Mongo` implementation. I also don't like leaving out `'` in names like this. `CanNotConnectToDatabaseError` I guess - but its preference not a big deal.

File: src/services/payment-flow/index.ts [hunk 3]
```diff
@@ -8,6 +8,7 @@ import {
 import { InvalidLightningPaymentFlowStateError, PaymentFlow } from "@domain/payments"
 import { paymentAmountFromNumber, WalletCurrency } from "@domain/shared"
 import { safeBigInt } from "@domain/shared/safe"
+import { parseRepositoryError } from "@services/mongoose/utils"
```
Comment (bodymindarts) [inline src/services/payment-flow/index.ts:11]: this file (ie `src/services/payment-flow`) should be under `mongoose` like all the other repositories.

File: src/services/redis/routes.ts [hunk 4]
```diff
@@ -16,7 +17,7 @@ export const RoutesCache = (): IRoutesCache => {
       await redis.set(key, value, "EX", defaultTimeToExpiryInSeconds)
       return routeToCache
     } catch (err) {
-      return new UnknownRepositoryError(err)
+      return parseRepositoryError(err)
```
Comment (bodymindarts) [inline src/services/redis/routes.ts:20]: The `redis` service should not be calling anything under `mongoose/utils` I don't see why the error messages should match anyway. `redis` service needs its own parse function.

File: src/services/ledger/services/transactions-metadata.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
-import { NoTransactionToUpdateError, UnknownRepositoryError } from "@domain/errors"
+import { NoTransactionToUpdateError } from "@domain/errors"
 import { CouldNotFindTransactionMetadataError } from "@domain/ledger"
-import { fromObjectId, toObjectId } from "@services/mongoose/utils"
+import { fromObjectId, toObjectId, parseRepositoryError } from "@services/mongoose/utils"
```
Comment (bodymindarts) [inline src/services/ledger/services/transactions-metadata.ts:3]: This isn't great but as this is also a repository implemented with `mongoose` its okay as an exception.

File: src/services/redis/routes.ts (hunk 4 above)
Comment (vindard) [reply src/services/redis/routes.ts]: Fixed

File: src/services/payment-flow/index.ts (hunk 3 above)
Comment (vindard) [reply src/services/payment-flow/index.ts]: Done: https://github.com/GaloyMoney/galoy/pull/1707/commits/a10d5f494d8e8f4ba12e260937ce5e9e261a03ca

File: src/domain/errors.ts (hunk 2 above)
Comment (vindard) [reply src/domain/errors.ts]: Done: https://github.com/GaloyMoney/galoy/pull/1707/commits/bd06ee7f110f2a85d1afdca9c34c64bda333b761

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**